openml/datasets/dataset.py

```diff
@@ -99,7 +101,8 @@ class OpenMLDataset(OpenMLBase):
         Serialized arff dataset string.
     """
     def __init__(self, name, description, format=None,
```

Comment:
I'm somewhat surprised that this one isn't annotated. @Neeratyoy could you please add to your stack to figure out why this is legal given that we have mypy running?

Reply:
Sure.
Just to confirm the task: I have to check why the missing annotation for _cache_format_ was never caught.

Reply:
Yes, that's correct.